src/conv/problem_description.cpp

```diff
@@ -67,15 +67,26 @@ void ProblemDescription::BuildConfKey(std::string& conf_key) const
     ss << 'x' << GetOutChannels();
     ss << 'x' << PrintDHW('x', GetSpatialDims(), GetOutDepth(), GetOutHeight(), GetOutWidth());
     ss << 'x' << GetInBatchSize();
-    ss << 'x' << GetInLayout();
+    if(GetDirection() != Direction::BackwardWeights)
+    {
+        ss << 'x' << GetInLayout();
+        ss << 'x' << GetWeightsLayout();
+        ss << 'x' << GetOutLayout();
+    }
+    else
+    {
+        ss << 'x' << GetInLayout();
+        ss << 'x' << GetOutLayout();
+        ss << 'x' << GetWeightsLayout();
+    }
+
     ss << 'x' << EncodeDataTypesForKey(GetInDataType(), GetWeightsDataType(), GetOutDataType());
     ss << 'x' << PrintDHW('x', GetSpatialDims(), GetPadD(), GetPadH(), GetPadW());
     ss << 'x'
        << PrintDHW(
               'x', GetSpatialDims(), GetKernelStrideD(), GetKernelStrideH(), GetKernelStrideW());
     ss << 'x' << PrintDHW('x', GetSpatialDims(), GetDilationD(), GetDilationH(), GetDilationW());
     ss << 'x' << GetGroupCount();
-    ss << 'x' << (GetDirection() == Direction::Forward ? "1" : "0");
```

Comment:
TODO: reenable this. Any objection for me to adopt the approach from `Serialize()` function?
`F`/`B`/`W` or `forward`/`backwarddata`/`backwardweights`

Reply:
Most likely , this is harmless, but I am not sure. Please discuss this with @DrizztDoUrden

Reply:
We actually need it when parsing the database files to fix swapped elements by the problem descriptor.

Reply:
Thanks for reminding, reverted to original 1/0

Reply:
> We actually need it when parsing the database files...

Database keys are produced by Serialize(), and direction is encoded there as F, B or W. NetworkConfig is used only in find-db, as a solution-specific, opaque payload.

Reply:
I think network confing string representation is only used in conjunction with algo name string containing direction. But it would be better to check that, I am not constantly looking through every change to the library.